### ISSUE TITLE
Idea: rename and make this a universal GraphQL fetcher

### ISSUE THREAD

**Issue by stubailo** [apollographql/apollo-network-interface#3]
If we could make a library that supports all this stuff like polling etc, it would be useful for:

1. Apollo client
2. Relay
3. GraphiQL
4. Someone just trying to fetch GraphQL from JavaScript

It would be cool if we designed it from the start to work for those cases. In that case I'd suggest renaming it to something like `apollo-fetcher` or `graphql-fetcher` or `apollo-network`?

Basically I think `network interface` is a janky name :P

**Comment by helfer** [comment]
Yeah, I think that's a good point. `graphql-fetcher` is already taken, but `apollo-fetcher` isn't.

**Comment by stubailo** [comment]
Oh - this doesn't include the transport? I thought it would. I do think we want a catchy name for this component + the transport, since that's something people could use as a lightweight non-caching client.

**Comment by thebigredgeek** [comment]
Bling bling
